### PR TITLE
Apply Network Policies Only in CIS Mode and Add Policy For DNS

### PR DESCRIPTION
**Pull request by briandowns**
#### Proposed Changes ####

* Adds a check to see if running in CIS mode and if so, apply the network policies.
* Adds a new network policy for DNS traffic.

#### Linked Issues ####

#297

### CODE REVIEW

File: pkg/rke2/np.go [hunk 1]
```diff
@@ -85,29 +133,78 @@ func setNetworkPolicy(ctx context.Context, namespace string, cs *kubernetes.Clie
...
+		ns.Annotations[namespaceAnnotationNetworkDNSPolicy] = cisAnnotationValue
+
+		if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+			if _, err := cs.CoreV1().Namespaces().Update(ctx, ns, metav1.UpdateOptions{}); err != nil {
+				if apierrors.IsConflict(err) {
+					if err := updateNamespaceRef(ctx, cs, ns); err != nil {
+						return err
+					}
+				}
+				return err
```
Comment (galal-hussein) [inline pkg/rke2/np.go:168]: in case of conflict this will still return error, should this be return updateNamespaceRef(...

**Comment by Oats87** [comment]
I think we should document that if you have these network policies set, ingress to namespaces other than kube-system will be broken.

**Comment by brandond** [comment]
Does allowing DNS ingress to coredns cover everything, or do other pods also need egress rules allowing this as well?

**Comment by briandowns** [comment]
> Does allowing DNS ingress to coredns cover everything, or do other pods also need egress rules allowing this as well?

I believe this is all that's needed.

**Review by brandond — APPROVED** [review]

**Review by galal-hussein — APPROVED** [review]
LGTM

**Pull request merged**